### ISSUE TITLE
BG2: web catches characters it shouldn't be able to

### ISSUE THREAD

**Issue by burner1024** [gemrb/gemrb#2026]
#### Bug description

Web has a save malus of -2 (also checked in spl file - with bg2fixpack), so any character with saving throw vs spells of -1 or better should be immune to it.

Here's a gnome illusionist with save vs spells of -1 getting webbed.

Additionally, the number of displayed saving throw rolls seems to vary from casting to casting. Example screenshots show 6 rolls, then 9. It should be 10 I think.

[save.zip](https://github.com/gemrb/gemrb/files/14393382/save.zip)

#### Steps to reproduce
1. Load game (alternatively, start a new ToB gnome illusionist with max Con, make sure to pick Web, and cheat yourself a Ring of Protection +2).
2. Cast Web, maybe a few times.
3. After a while, PC will get webbed.

#### Expected behavior

Any character with saving throw vs spells of -1 or better should be immune to Web.

#### Screenshots
![Captura desde 2024-02-24 22-05-42](https://github.com/gemrb/gemrb/assets/1279852/dba4f853-5cab-44fb-a393-e9a5f0c3f0e2)
![Captura desde 2024-02-24 22-19-52](https://github.com/gemrb/gemrb/assets/1279852/29e0435f-3b14-40b8-8f45-355d9a2d3d48)
![Captura desde 2024-02-24 22-20-14](https://github.com/gemrb/gemrb/assets/1279852/ce926557-108b-4609-a39d-1a9b641d0233)
![Captura desde 2024-02-24 22-07-25](https://github.com/gemrb/gemrb/assets/1279852/964777d6-ac82-482f-a1ca-c04ca90dbd9e)
![Captura desde 2024-02-24 22-21-27](https://github.com/gemrb/gemrb/assets/1279852/6e9df0d8-007f-4d93-b9a7-fcab42d1ecc3)
![Captura desde 2024-02-24 22-21-38](https://github.com/gemrb/gemrb/assets/1279852/0c89cf45-2940-40f5-a061-51d600a76a6c)

GemRB version (check as many as you know apply)
- [x] master as of this issue
- [ ] 0.9.2
- [ ] 0.9.1
- [ ] 0.9.0

Video Driver (check as many as you know apply)
- [ ] SDL1.2
- [ ] SDL2 built with `OPENGL_BACKEND` enabled
- [ ] SDL2 without `OPENGL_BACKEND` enabled

**Comment by lynxlynxlynx** [comment]
Definitely not true for evokers (both as caster and targets), which is web's school.
If you want to investigate, `CheckSaves` is the place to break on, just make sure it's the right opcode (you can check `fx->Opcode`).

**Comment by burner1024** [comment]
It seems to hit when `ret=1`, or 19/20.
https://github.com/gemrb/gemrb/blob/253e06cac9eb8247e101235198442278bfda422d/gemrb/core/Scriptable/Actor.cpp#L3156-L3159

Saving throw rolls do not have critical hits nor failures. Not in 2.5, at least.

**Comment by lynxlynxlynx** [comment]
Scouring the net agrees that there are no critical misses in bg. There was one suggestions that there are critical hits, but at 20 that could easily be incidental. There was one suggestion it is different in iwd2. All the rest say nothing, iwd2re isn't complete here yet, so I think it's safe to remove without further testing.

**Comment by burner1024** [comment]
This is also wrong
https://github.com/gemrb/gemrb/blob/253e06cac9eb8247e101235198442278bfda422d/gemrb/core/Scriptable/Actor.cpp#L3186
The [rule](https://adnd2e.fandom.com/wiki/The_Saving_Throw_(DMG)) is to roll greater or equal to save. In 3rd ed. [too](https://www.dandwiki.com/wiki/3e_SRD:Saving_Throw).

I think this could be a good opportunity to add roll breakdown for #1800. At least, for the second edition part. Controlled by a new var in GemRB.cfg maybe?

**Comment by lynxlynxlynx** [comment]
Quoting the SRD is pointless, since these are simplified games and even their manuals are unreliable. But the relation is easy to test anyway.

The saves part of #1800 would have its printing here, but we don't save enough data if you want to see all the modifiers.

**Comment by burner1024** [comment]
> Quoting the SRD is pointless, since these are simplified games and even their manuals are unreliable. But the relation is easy to test anyway.

I know this for sure, just like [Stoneskin not blocking spells](#1801). These are fundamental game rules.
At least in BG2. I'm 95% sure it's the same in other games, and even if it wasn't, I would say that it's a bug that should be fixed anyway.

> The saves part of https://github.com/gemrb/gemrb/issues/1800 would have its printing here, but we don't save enough data if you want to see all the modifiers.

Well, can I add printing whatever is possible for now or do you want to postpone this?

**Comment by lynxlynxlynx** [comment]
We try to avoid adding special casing if it's not confirmed to be needed, that's why testing is so useful.

The feature request should be discussed in its own channel, but a cfg option definitely isn't the way to go, a custom strref is.

**Comment by burner1024** [comment]
> We try to avoid adding special casing if it's not confirmed to be needed, that's why testing is so useful.

Sorry, what do you mean by special casing?

**Comment by lynxlynxlynx** [comment]
Eg. just changing the equation for bg2.

**Comment by burner1024** [comment]
I don't propose any special casing, I think it should be applied to all games.

**Comment by lynxlynxlynx** [comment]
And I'm saying it should be tested first.

**Comment by burner1024** [comment]
To which I suggested that even it's not the case, that would be a bug, and all the more reason to fix it.

Why do I think it would be a bug? Well... DMG is pointless, manuals are wrong, what would be the authoritative source, original programmers? Here's a [video](https://youtu.be/zfmekgRfHkM?t=413) of Mark Darrah talking how craftily he implented Timestop to pause all effects, nothing ticking and expiring. I decided not to tell him that he blew it :)...
At this point, I'm sure that people actively involved with the games are more equipped to answer that. Current implementation has no argument to support it.
